### DISCUSSION TITLE
Advanced Foundry | NFT Lesson | Testnet Deployment

### DISCUSSION THREAD

**Discussion by vigneshbalu96** [Cyfrin/foundry-full-course-cu#3067] [Q&A]
:hand:  Hello Blockchianeers, while running the command while deploying the PUG NFT, getting error.

` $ make deploy ARGS="--network sepolia" `
error: a value is required for '--private-key <RAW_PRIVATE_KEY>' but none was supplied

For more information, try '--help'.
make: *** [Makefile:40: deploy] Error 2

and I modified my Makefile with --account <wallet_Account_Name>. Modified the script to avoid keeping private key in plain text.

`deploy:
	@forge script script/DeployBasicNft.s.sol:DeployBasicNft $(NETWORK_ARGS) --account sepoliaAccount`
	
what would be a better solution to this? :thinking:

**Comment by EngrPips** [comment]
were you able to deploy with this `deploy: @forge script script/DeployBasicNft.s.sol:DeployBasicNft $(NETWORK_ARGS) --account sepoliaAccount`?

**Reply by vigneshbalu96** [reply]
@EngrPips Nope!, I'm curios to know how we can work with that without revealing the private key in plain text in the `.env` file while working with testnet deployment.

**Reply by EngrPips** [reply]
I think that is what Patrick explained in this video https://www.youtube.com/watch?v=VQe7cIpaE54
